Firestore/core/src/model/target_index_matcher.cc

```diff
@@ -36,15 +36,11 @@ TargetIndexMatcher::TargetIndexMatcher(const core::Target& target) {
                        ? *target.collection_group()
                        : target.path().last_segment();
   order_bys_ = target.order_bys();
-  inequality_filter_ = absl::nullopt;
 
   for (const Filter& filter : target.filters()) {
     FieldFilter field_filter(filter);
     if (field_filter.IsInequality()) {
-      HARD_ASSERT(!inequality_filter_.has_value() ||
-                      inequality_filter_->field() == field_filter.field(),
-                  "Only a single inequality is supported");
-      inequality_filter_ = field_filter;
+      inequality_filters_.emplace(field_filter);
```

Comment:
nit: I'd recommend to just use `insert` instead of `emplace` since there probably isn't much benefit to using emplace (if any) and it ever-so-slightly detracts from readability IMO due to this lack of a strong motivation to use emplace(). Moreover, the "else" block uses push_back(), which is inconsistent. Either both branches should emplace or both should copy. Ideally, both should just insert/push with move to avoid the copy.